index.js

```diff
@@ -918,3 +918,16 @@ module.exports = input => {
 };
 
 Object.defineProperty(module.exports, 'minimumBytes', {value: 4100});
+
+module.exports.stream = readableStream => new Promise(resolve => {
+	const stream = readableStream;
+
+	function helper(stream) {
+		const chunk = stream.read(module.exports.minimumBytes);
+		stream.fileType = module.exports(chunk);
+		stream.unshift(chunk);
+		resolve(stream);
+	}
+
+	stream.once('readable', () => helper(stream));
+});
```

Comment:
This could be written simpler without the `helper` function:

```js
module.exports.stream = readableStream => new Promise(resolve => {
	readableStream.once('readable', () => {
		const chunk = readableStream.read(module.exports.minimumBytes);
		readableStream.fileType = module.exports(chunk);
		readableStream.unshift(chunk);
		resolve(readableStream);
	});
});
```

Reply:
Agreed. It was initially there because of the `.removeListener` method. With `.once` it's not necessary.